packages/core/core/src/requests/PathRequest.js

```diff
@@ -220,6 +221,10 @@ export class ResolverRunner {
             };
           }
 
+          if (result.isAsync != null) {
+            dependency.isAsync = result.isAsync;
```

Comment:
Not totally sure how I feel about mutating the dependency in the resolver, but we did it already for meta... Otherwise we'd need to move `isAsync` to the AssetGroup, but that's not available in the bundle graph. 😞